src/backend/joanie/signature/backends/lex_persona.py

```diff
@@ -571,3 +571,63 @@ def get_signed_file(self, reference_id: str) -> bytes:
             )
 
         return response.content
+
+    def _prepare_update_of_signatories(self, reference_id: str) -> dict[dict]:
+        """
+        Prepare the payload to update an ongoing signature procedure.
+        """
+        validity_period = settings.JOANIE_SIGNATURE_VALIDITY_PERIOD
+        order = models.Order.objects.get(
+            contract__signature_backend_reference=reference_id
+        )
+        organization_recipient_data = (
+            self._prepare_recipient_data_for_organization_signer(order)
+        )
+        return {
+            "steps": [
+                {
+                    "stepType": "signature",
+                    "recipients": organization_recipient_data,
+                    "requiredRecipients": 1,
+                    "validityPeriod": validity_period,
+                    "invitePeriod": None,
+                    "maxInvites": 0,
+                    "sendDownloadLink": True,
+                    "allowComments": False,
+                    "hideAttachments": False,
+                    "hideWorkflowRecipients": False,
+                },
+            ]
```

Comment:
If I understand well what you did after this update there will be only one step in the workflow. Am I right?


If yes, I don't think that's what we want because : 
- If the student has not yet signed its contract, it will not be able to sign anymore ?
- Will the student be able to download its contract if it has been removed from the workflow ?

I just wonder one thing if we have to redeclare all steps : 
- If the student has already signed its contract, does it have to sign again ?

Reply:
**When the student has not signed:**

- we update all the signatories

**When the student has signed already :**
- we update the organization signatories only

What happened while testing with the signature provider API, when the student has already signed the contract and I update every signatories, it will create a 3 steps instead of 2 for signing the document. The student will need to sign two times. Which is not accurate and it requires  3 signatures to validate the contract to a finished event for our webhook.